### PR TITLE
Added Slime view engine with test

### PR DESCRIPTION
**Pull request by robconery**
Not sure if this is important to you but I really like the Slime view engine so I decided to add it in. I wanted to open a PR to work against... so here you go! I'm not completely convinced I have it right; I used the PhoenixSlime project as a guide for plugging in a template engine with Slime, so hopefully it works as intended. 

I added a test as well, following your convention.

### CODE REVIEW

File: lib/trot/template/slim.ex [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  @behaviour Trot.Template.Engine
+
+  @doc false
+  def compile(file) do
+    quote do
+      unquote(file)
+      |> File.read!
+      |> Slime.Renderer.precompile 
+      |> EEx.compile_string assigns: var!(assigns)
```
Comment (hexedpackets) [inline lib/trot/template/slim.ex]: The `compile/1` function doesn't work. `EEx.compile_string/2` returns a quoted expression, which you're then double-quoting in the compile function. I think what you can do here is use `Slime.Renderer.render/2` in place of botht he Slime and EEx functions (and keep the `quote` block).

Comment (robconery) [reply lib/trot/template/slim.ex]: That was the one I was having trouble with - I couldn't figure out where/how that function was ever called or used. Is there a test somehow/somewhere?

[the remark this answers is not in the document]
Comment (robconery) [reply lib/trot/template/slim.ex]: Well... that's embarrassing. Elixir is a bit rusty :) ...

**Comment by robconery** [comment]
Alright - this should do the trick. I do get one failure on the Auth test but it doesn't seem to get triggered by the build so I'm assuming it's something local. The test I added is green.

File: lib/trot/template/slim.ex (hunk 1 above)
Comment (hexedpackets) [reply lib/trot/template/slim.ex]: I never figured out a good way to test this. Its used by the `Trot.Template` macros based on the mix environment - dev uses `compile/2` while other environments use `full_compile/2`. The difference is whether the template is precompiled in memory for faster responses or read off disk every time to allow for live reloading during development.

**Pull request merged**